Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1393,12 +1397,43 @@ void LocalFrame::networkUsageDidExceedThreshold()
 
     FRAME_RELEASE_LOG_ERROR(ResourceLoading, "networkUsageDidExceedThreshold: Unloading frame due to exceeding threshold.");
 
-    showResourceMonitoringError("This frame was unloaded."_s);
+    showResourceMonitoringError(generateResourceMonitorErrorHTML());
 
     if (RefPtr document = this->document())
         document->addConsoleMessage(MessageSource::ContentBlocker, MessageLevel::Error, "Frame was unloaded because its network usage exceeded the limit."_s);
 }
 
+String LocalFrame::generateResourceMonitorErrorHTML() const
```

Comment:
nit: seems like this could just be a `static` function in this file and doesn't need to touch `LocalFrame.h`.